datascience/tables.py

```diff
@@ -395,17 +395,156 @@ def append_column(self, label, values):
         self._columns[label] = values
         return self
 
+    def with_column(self, label, values):
+        """Returns a separate new table with new column included.
+
+        Args:
+            ``label`` (str): The label of the new column.
+                Must be a string.
+
+            ``values`` (single value or list/array): If a single value, every
+                value in the new column is ``values``.
+
+                If a list or array, the new column contains the values in
+                ``values``. ``values`` must be the same length as the table.
+
+        Returns:
+            A new separate table containing new column specified by ``label``
+            and ``values`` in addition to previous columns.
+
+        Raises:
+            ``ValueError``: If
+                - ``label`` is not a string.
+                - ``values`` is a list/array and does not have the same length
+                  as the number of rows in the table.
+
+        >>> table
+        letter | count | points
+        a      | 9     | 1
+        b      | 3     | 2
+        c      | 3     | 2
+        z      | 1     | 10
+
+        >>> table2 = table.append_column('new_col1', [10, 20, 30, 40])
+        >>> table3 = table2.append_column('new_col2', 'hello')
+        >>> table2
+        letter | count | points | new_col1
+        a      | 9     | 1      | 10
+        b      | 3     | 2      | 20
+        c      | 3     | 2      | 30
+        z      | 1     | 10     | 40
+        >>> table3
+        letter | count | points | new_col1 | new_col2
+        a      | 9     | 1      | 10       | hello
+        b      | 3     | 2      | 20       | hello
+        c      | 3     | 2      | 30       | hello
+        z      | 1     | 10     | 40       | hello
+        >>> table
+        letter | count | points
+        a      | 9     | 1
+        b      | 3     | 2
+        c      | 3     | 2
+        z      | 1     | 10
```

Comment:
Good that you showed that `table` is unchanged! But you wrote `table.append_column('new_col1', [10, 20, 30, 40])` instead of `table.with_column('new_col1', [10, 20, 30, 40])`. :)